lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/impl/MultiObserverImpl.java

```diff
@@ -36,20 +36,19 @@
 public abstract class MultiObserverImpl<TSubCls> implements MultiObserver {
   private static final Logger LOG = Logger.getLogger(MultiObserverImpl.class.getName());
   private final Map<String, Method> methodMap = new HashMap<>();
-  private boolean initialized = false;
 
   /**
    * Use reflection to discover all of the event processing methods in TSubCls
    * and setup a means to direct calls from the generic event onNext method defined
    * in the MultiObserver interface to specific concrete event onNext methods.
    */
-  private void initialize() {
+  public MultiObserverImpl() {
     // Iterate across the methods and build a hash map of class names to reflection methods.
     for (final Method method : this.getClass().getMethods()) {
       if (method.getName().equals("onNext") && method.getDeclaringClass().equals(this.getClass())) {
         // This is an onNext method defined in TSubCls
         final Class<?>[] types = method.getParameterTypes();
-        if (types.length == 2 && types[0].getSimpleName() == "long") {
+        if (types.length == 2 && types[0].getSimpleName().equals("long")) {
```

Comment:
Actually, a better way to check if the parameter is of the right type would be something like `types[0].equals(Long.TYPE)` (although `==` should be fine in that particular case, too, I guess).